### PR TITLE
graph/encoding/dot: don't panic when an Attributers returns a nil

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

### CODE REVIEW

**Comment by mattConn** [comment]
Would it be of use to have a test case involving a graph with at least one nil attributer to demonstrate this behavior? Otherwise, it looks good to me.

**Review by mattConn — APPROVED** [review]

**Comment by kortschak** [comment]
Added test case; only needed modification of existing tests.

**Review by mattConn — APPROVED** [review]

File: graph/encoding/dot/encode_test.go [hunk 1]
```diff
@@ -1183,13 +1183,14 @@ var encodeTests = []struct {
...
-			graph: []encoding.Attribute{{Key: "rankdir", Value: `"LR"`}},
-			node:  []encoding.Attribute{{Key: "fontsize", Value: "16"}, {Key: "shape", Value: "ellipse"}},
+		g: graphAttributer{
+			Graph: undirectedEdgeAttrGraphFrom(powerMethodGraph, map[edge][]encoding.Attribute{
+				{from: 0, to: 2}: {{Key: "label", Value: `"???"`}, {Key: "style", Value: "dashed"}},
+				{from: 2, to: 4}: {},
+				{from: 3, to: 4}: {{Key: "color", Value: "red"}},
+			}),
+			graph: attributer{{Key: "rankdir", Value: `"LR"`}},
+			node:  attributer{{Key: "fontsize", Value: "16"}, {Key: "shape", Value: "ellipse"}},
```
[the remark this answers is not in the document]
Comment (kortschak) [reply graph/encoding/dot/encode_test.go:1193]: You are a hard task master.

Done.

**Comment by kortschak** [comment]
PTAL

**Review by mattConn — APPROVED** [review]
LGTM!

**Pull request merged**